src/Containers/Microsoft.NET.Build.Containers/AuthHandshakeMessageHandler.cs

```diff
@@ -166,6 +166,11 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             request.Headers.Authorization = cachedAuthentication;
         }
 
+        if (request.Method == HttpMethod.Patch && request.Content?.Headers.ContentLength > 0)
+        {
+            request.Headers.AcceptEncoding.Add(new StringWithQualityHeaderValue("gzip"));
```

Comment:
should we set this on the atomic-upload request specifically instead of in the auth handler?

Reply:
it was fast way to do it.  It is also needed for chunks, though no way to prove now. 
Since we are using single `HttpClient` everywhere, we need to move to `SendAsync` to be able to set the headers, it needs a bit of rework.

Reply:
Changed this